src/query/ts/m3db/consolidators/step_consolidator.go

```diff
@@ -63,7 +63,7 @@ func NewStepLookbackConsolidator(
 	startTime time.Time,
 	fn ConsolidationFunc,
 ) *StepLookbackConsolidator {
-	datapoints := make([]ts.Datapoint, 0, initLength)
+	datapoints := make([]ts.Datapoint, 0, initLength*BufferSteps)
```

Comment:
Hm, should this perhaps just be `BufferSteps` instead of `initLength*BufferSteps`? 

From my current perhaps limited understanding we should only ever be buffering up to `BufferSteps` max in `c.unconsumed` so having 320 values instead of 32 values (initLength=10, BufferSteps=32) able to be buffered might be a bit of a waste given `*StepLookbackConsolidator` should not need more than 32?